src/lang/base/BuiltIns.ml

```diff
@@ -1059,91 +1059,91 @@ module ScillaBuiltIns
-      ("sub", Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.sub);
-      ("mul", Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.mul);
-      ("div", Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.div);
-      ("rem", Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.rem);
-      ("pow", Uint.pow_arity, Uint.pow_type, Uint.pow_elab, Uint.pow);
-      ("to_uint32", Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits32, Uint.to_uint32);
-      ("to_uint64", Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits64, Uint.to_uint64);
-      ("to_uint128", Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits128, Uint.to_uint128);
-      ("to_uint256", Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits256, Uint.to_uint256);
-      ("to_nat", Uint.to_nat_arity, Uint.to_nat_type, Uint.to_nat_elab, Uint.to_nat);
+      (Builtin_eq, Uint.eq_arity, Uint.eq_type, Uint.eq_elab, Uint.eq);
+      (Builtin_lt, Uint.eq_arity, Uint.eq_type, Uint.eq_elab, Uint.lt);
+      (Builtin_add, Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.add);
+      (Builtin_sub, Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.sub);
+      (Builtin_mul, Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.mul);
+      (Builtin_div, Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.div);
+      (Builtin_rem, Uint.binop_arity, Uint.binop_type, Uint.binop_elab, Uint.rem);
+      (Builtin_pow, Uint.pow_arity, Uint.pow_type, Uint.pow_elab, Uint.pow);
+      (Builtin_to_uint32, Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits32, Uint.to_uint32);
+      (Builtin_to_uint64, Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits64, Uint.to_uint64);
+      (Builtin_to_uint128, Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits128, Uint.to_uint128);
+      (Builtin_to_uint256, Uint.to_uint_arity, Uint.to_uint_type, Uint.to_uint_elab Bits256, Uint.to_uint256);
+      (Builtin_to_nat, Uint.to_nat_arity, Uint.to_nat_type, Uint.to_nat_elab, Uint.to_nat);
     ]
 
     let built_in_hashtbl =
       let open Caml in
-      let ht : ((string, built_in_record list) Hashtbl.t) = Hashtbl.create 64 in
+      let ht : ((builtin, built_in_record list) Hashtbl.t) = Hashtbl.create 64 in
       List.iter (fun row ->
           let (opname, _, _, _, _) = row in
           match Hashtbl.find_opt ht opname with
           | Some p ->  Hashtbl.add ht opname (row::p)
           | None -> Hashtbl.add ht opname [row]
         ) built_in_dict;
       ht
-      
+
     (* Dictionary lookup based on the operation name and type *)
-    let find_builtin_op op argtypes =
-      let opname = get_id op in
-      let finder = (function (name, arity, optype, elab, exec) ->
-          if name = opname && arity = List.length argtypes
+    let find_builtin_op b argtypes =
+      let TaggedBuiltin (op, rep) = b in
+      let finder = (function (builtin, arity, optype, elab, exec) ->
```

Comment:
I see that the build now fails due to the unused variable `builtin`, so it should be replaced with a wildcard (`_`).

Reply:
Prepending variable names with an underscore also works, I'd prefer to keep the name in the form of `_builtin`.